scu-de/src/fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2/SwissbitCloudV2SCU.cs

```diff
@@ -0,0 +1,551 @@
+                {
+                    return request;
+                }
+                if (tssResult.InitializationState == SwissbitCloudV2TseState.uninitialized.ToString())
+                {
+                    throw new SwissbitCloudV2Exception("The state of the TSE is 'UNINITIALIZED' and therefore not yet personalized, which is currently not supported.");
+                }
+                if (request.CurrentState != TseStates.Terminated)
+                {
+                    throw new SwissbitCloudV2Exception($"The state of the TSE is {tssResult.InitializationState} and therefore this request is not supported.");
+                }
+
+                var tseResult = await _swissbitCloudV2Provider.DisableTseAsync();
+
+                return ((SwissbitCloudV2TseState) Enum.Parse(typeof(SwissbitCloudV2TseState), tseResult.InitializationState, true)).ToTseState();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to execute {Operation} - Request: {Request}", nameof(SetTseStateAsync), JsonConvert.SerializeObject(request));
+                throw;
+            }
+        }
+
+        public Task ExecuteSetTseTimeAsync() => Task.CompletedTask;
+
+        public Task ExecuteSelfTestAsync() => Task.CompletedTask;
+
+        public async Task<StartExportSessionResponse> StartExportSessionAsync(StartExportSessionRequest request)
+        {
+            try
+            {
+                if (!_configuration.EnableTarFileExport)
+                {
+                    return new StartExportSessionResponse
+                    {
+                        TokenId = _noExport + Guid.NewGuid().ToString(),
+                        TseSerialNumberOctet = _configuration.TseSerialNumber
+                    };
+                }
+
+                var startExportResponse = await _swissbitCloudV2Provider.StartExport();
+
+                CacheExportAsync(startExportResponse.ExportId).ExecuteInBackgroundThread();
+
+                SetExportState(startExportResponse.ExportId, ExportState.Running);
+
+                return new StartExportSessionResponse
+                {
+                    TokenId = startExportResponse.ExportId,
+                    TseSerialNumberOctet = _configuration.TseSerialNumber
+                };
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to execute {Operation} - Request: {Request}", nameof(StartExportSessionAsync), JsonConvert.SerializeObject(request));
+                throw;
+            }
+        }
+
+        private async Task CacheExportAsync(string exportId, int currentTry = 0)
+        {
+            try
+            {
+                await _swissbitCloudV2Provider.StoreDownloadResultAsync(exportId);
+                SetExportState(exportId, ExportState.Succeeded);
+            }
+            catch (WebException)
```

Comment:
Do we know if the Swissbit API will throw a WebException during exports? To my understanding, this is thrown when something goes wrong on "network level" 🤔 

What was the case we had with fiskaly, again? I can't remember anymore right now

Reply:
That's the way we do it with Fiskaly we retry according to retry value in the config.